admin/services/tech-support/ps-report-logs.service/data-access/ps-report-logs.data.service.ts

```diff
@@ -0,0 +1,41 @@
+
+import { BlobServiceClient } from '@azure/storage-blob'
+import { familiarisationCheckEndDateAfterAdminEndDate } from 'lib/errors/check-window-v2'
+import config from '../../../../config'
+
+export interface IPsReportLogsDataService {
+  getContainerList (): Promise<Array<string>>
+}
+
+export class PsReportLogsDataService {
+  public static async getContainerList (): Promise<Array<string>> {
+    const client = BlobServiceClient.fromConnectionString(config.AZURE_STORAGE_CONNECTION_STRING)
+    const list = new Array<string>()
+    for await (const container of client.listContainers({ includeMetadata: false })) {
+      list.push(container.name)
+    }
+    return list
+  }
+
+  public static async getFileContents (containerName: string, logFileName: string): Promise<string | undefined> {
+    const client = BlobServiceClient.fromConnectionString(config.AZURE_STORAGE_CONNECTION_STRING)
+    const containerClient = await client.getContainerClient(containerName)
+    if (!containerClient.exists()) return
+    const blobClient = containerClient.getBlobClient(logFileName)
+    if (!blobClient.exists()) return
+    // 1GB limit on 32bit, 2GB limit on 64bit...
+    const blobBuffer = await blobClient.downloadToBuffer()
+    return blobBuffer.toString()
+  }
+
+  public static async getContainerFileList (containerName: string): Promise<string[]> {
+    const client = BlobServiceClient.fromConnectionString(config.AZURE_STORAGE_CONNECTION_STRING)
+    const containerClient = await client.getContainerClient(containerName)
+    if (!containerClient.exists()) return []
+    const files = new Array<string>()
+    for await (const file of containerClient.listBlobsFlat({ includeMetadata: false, includeDeleted: false })) {
```

Comment:
I am surprised we are not getting the metadata.  Does this mean you are not showing the file size to the user next to the download link?

Reply:
yes that's my plan.  i realise now i've not put this up as a draft 🤦🏼